src/modules/AutoinstClone.rb

```diff
@@ -98,88 +93,49 @@ def CommonClone(_resource, resourceMap)
-                Ops.get_string(resourceMap, "Name", "")
-              )
-            name = Ops.get_string(resourceMap, "Name", "")
-          end
-          # Set resource name, if not using default value
-          resource = Ops.get_string(
-            resourceMap,
-            "X-SuSE-YaST-AutoInstResource",
-            ""
-          )
-          resource = def_resource if resource == ""
-          items = if resource != ""
-            Builtins.add(items, Item(Id(resource), name))
-          else
-            Builtins.add(items, Item(Id(def_resource), name))
-          end
-        end
-      end
-      # sort items for nicer display
-      items = Builtins.sort(
-        Convert.convert(items, from: "list", to: "list <term>")
-      ) do |x, y|
-        Ops.less_than(Ops.get_string(x, 1, "x"), Ops.get_string(y, 1, "y"))
+      module_map = Y2ModuleConfig.ModuleMap
+      clonable_items = module_map.each_with_object([]) do |(def_resource, resource_map), items|
+        log.debug "r: #{def_resource} => #{resource_map["X-SuSE-YaST-AutoInstClonable"]}"
+        clonable = resource_map["X-SuSE-YaST-AutoInstClonable"] == "true"
+        next unless clonable
+
+        desktop_file = resource_map.fetch("X-SuSE-DocTeamID", "")[4..]
+        translation_key = "Name(#{desktop_file}.desktop): #{resource_map["Name"]}"
+        name = Builtins.dpgettext("desktop_translations", "/usr/share/locale/", translation_key)
+        name = resource_map.fetch("Name", "") if name == translation_key
+        # Set resource name, if not using default value
+        resource_name = resource_map.fetch("X-SuSE-YaST-AutoInstResource", "")
+        resource_name = def_resource if resource_name.empty?
+        items << Item(Id(resource_name), name)
       end
-      deep_copy(items)
+
+      clonable_items.sort_by { |i| i[1] }
     end
 
-    # Build the profile
-    # @return [void]
+    # Builds the profile
+    #
+    # @return [nil]
+    # @see Profile.Prepare
     def Process
-      log.info "Base resources: #{@base} additional: #{@additional}"
+      log.info "Additional resources: #{@additional}"
       Profile.Reset
       Profile.prepare = true
       Mode.SetMode("autoinst_config")
 
-      Builtins.foreach(Y2ModuleConfig.ModuleMap) do |def_resource, resourceMap|
+      Y2ModuleConfig.ModuleMap.each do |def_resource, resource_map|
         # Set resource name, if not using default value
-        resource = Ops.get_string(
-          resourceMap,
-          "X-SuSE-YaST-AutoInstResource",
-          ""
-        )
-        resource = def_resource if resource == ""
-
-        if @additional.include?(resource)
-          log.info "Now cloning: #{resource}"
-          time_start = Time.now
-          CommonClone(def_resource, resourceMap)
-          log.info "Cloning #{resource} took: #{(Time.now - time_start).round} sec"
-        end
+        resource = resource_map.fetch("X-SuSE-YaST-AutoInstResource", "")
+        resource = def_resource if resource.empty?
```

Comment:
I see this logic multiple times, can we abstract it to own method?

Reply:
You are right. However, I preferred to keep as it is by now because I think we should rework that part completely. Having to deal with a hash keyed by "X-SuSE-YaST-XXX" strings is painful. We have two related Trello cards: [this](https://trello.com/c/VZ06T8Ov/7-research-rethink-the-modules-registry) and [this](https://trello.com/c/mxigZ6UJ/47-rework-way-autoyast-sections-relate-to-required-packages).